### PR TITLE
[kineto] Update Kineto submodule, cupti library paths

### PR DESCRIPTION
**Pull request by ilia-cher**
Summary:
Update kineto submodule, improve cupti detection

Test Plan: CI

### CODE REVIEW

**Comment by ilia-cher** [comment]
this should also help with https://github.com/pytorch/pytorch/issues/56890 issue

**Review by malfet — APPROVED** [review]

File: caffe2/python/__init__.py [hunk 1]
```diff
@@ -37,12 +37,14 @@
         cuda_path_var = 'CUDA_PATH_V' + cuda_version_1
         default_path = 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v' + cuda_version
         cuda_path = os.path.join(os.getenv(cuda_path_var, default_path), 'bin')
+        cuda_extras_path = os.path.join(os.getenv(cuda_path_var, default_path), 'extras', 'CUPTI', 'lib64')
```
Comment (malfet) [inline caffe2/python/__init__.py:40]: Is that a correct path on Windows platform?

Comment (ilia-cher) [reply caffe2/python/__init__.py:40]: (discussed)

**Comment by ngimel** [comment]
Is there a reason we are not using FindCUDAToolkit cupti target https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cuda-toolkit-cupti or CUDA_cupti_LIBRARY set by FindCUDA  https://cmake.org/cmake/help/latest/module/FindCUDA.html?

**Comment by ilia-cher** [comment]
> Is there a reason we are not using FindCUDAToolkit cupti target https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cuda-toolkit-cupti or CUDA_cupti_LIBRARY set by FindCUDA https://cmake.org/cmake/help/latest/module/FindCUDA.html?

I tried using CUDA_cupti_LIBRARY, the issue was that I couldn't control what type of library I wanted: dynamic or static

**Comment by ilia-cher** [comment]
> Is there a reason we are not using FindCUDAToolkit cupti target https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cuda-toolkit-cupti or CUDA_cupti_LIBRARY set by FindCUDA https://cmake.org/cmake/help/latest/module/FindCUDA.html?

though I see now that FindCUDAToolkit exports `CUDA::cupti_static` - will try that, thanks!

**Comment by ilia-cher** [comment]
> 
> though I see now that FindCUDAToolkit exports `CUDA::cupti_static` - will try that, thanks!

unfortunately FindCUDAToolkit seems to be available only since 3.17 cmake

**Pull request closed without merging**